src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -719,6 +624,90 @@ public CompilationLoader(SourceLoader loadSources, IEnumerable<string> reference
         {
         }
 
+        private IEnumerable<(LoadedStep, Action<Status>)> InternalRewriteSteps(Uri executingAssembly)
+        {
+            // TODO: The dependencies for rewrite steps should be declared as part of IRewriteStep interface, and we
+            // should query those here.
+#pragma warning disable CS0618 // Type or member is obsolete
+            var dependencies = BuiltIn.AllBuiltIns.Select(b => b.FullName);
+#pragma warning restore CS0618 // Type or member is obsolete
+
+            var status = this.compilationStatus;
+            var steps = new (IRewriteStep Step, bool Enabled, Action<Status> SetStatus)[]
+            {
+                // TODO: It would be nicer to trim unused intrinsics. Currently, this is not possible due to how the

```

Comment:
Same here, it would be great to create an issue and reference it in the comment to not lose track of it.

Reply:
Same as above, this is not a new todo.